src/updates/node/index.ts

```diff
@@ -124,10 +115,10 @@ export function getReleaseNotes(version: string): string {
 
 }
 
-export async function checkForUpdate(): Promise<UpdateInfo> {
+export async function checkForUpdate(supportedVersions?: string): Promise<UpdateInfo> {
 	const [ currentVersion, latestVersion ] = await Promise.all<string | undefined, string>([
 		checkInstalledVersion(),
-		checkLatestVersion()
+		checkLatestVersion(supportedVersions)
```

Comment:
The return value of `checkLatestVersion` conflicts with the undefined stated in the type in line 119.

I think maybe it's best rather than returning `semver.clean(latestVersion);` directly, get the value and check it, if it's null throw an error, otherwise return the string we get from that. Then the return type for `checkLatestVersion` would be `Promise<string>` and be fine with the above typing